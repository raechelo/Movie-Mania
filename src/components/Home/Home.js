import React from 'react';
import { connect } from 'react-redux';

const Home = (props) => {
const movieImage = props.movies.length && props.movies.map(m => (m.title && <img src={m.posterImage} key={m.id} alt="poster" />))

  return (
    <section className='Home'>{movieImage}</section>
  )
}

export const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(Home);