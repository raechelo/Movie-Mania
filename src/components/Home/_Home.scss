h2 {
  display: block;
  margin: auto;
  font-family: $fancyfont;
}