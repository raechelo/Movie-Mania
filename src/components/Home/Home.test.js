import React from 'react';
import { shallow } from 'enzyme';
import { Home } from './Home';

describe('Home', () => {

  let wrapper;

  let movies = [{title: 'Back to the Future'}, {title: 'Braveheart'}]

  beforeEach(() => {
    wrapper = shallow( <Home movies={movies} /> )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
})