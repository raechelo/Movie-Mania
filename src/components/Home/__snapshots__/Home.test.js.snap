// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot with all the data passed in 1`] = `
<section
  className="Home"
>
  <Link
    to="/movies/undefined"
  >
    <ConnectFunction
      m={
        Object {
          "title": "Back to the Future",
        }
      }
    />
     
  </Link>
  <Link
    to="/movies/undefined"
  >
    <ConnectFunction
      m={
        Object {
          "title": "Braveheart",
        }
      }
    />
     
  </Link>
</section>
`;
